frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1412,18 +1415,32 @@ impl<T: Config> Pallet<T> {
 	/// Extracted for easier weight calculation.
 	fn create_snapshot_external(
 	) -> Result<(Vec<T::AccountId>, Vec<VoterOf<T>>, u32), ElectionError<T>> {
-		let target_limit = T::MaxElectableTargets::get().saturated_into::<usize>();
-		let voter_limit = T::MaxElectingVoters::get().saturated_into::<usize>();
-
-		let targets = T::DataProvider::electable_targets(Some(target_limit))
+		let election_bounds = T::ElectionBounds::get();
+		let target_limit = election_bounds
+			.targets
+			.count
```

Comment:
```suggestion
			.count.map_or(usize::MAX, |c| c.0.saturated_into::<usize>())
```

Reply:
@kianenigma do you think `usize::MAX` is a good idea? I'm kinda afraid it's going to be too large.